### PR TITLE
Improve resource quota doc.

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

File: docs/admin/resource-quota.md [hunk 1]
```diff
@@ -31,12 +31,30 @@ Documentation for other releases can be found at
...
+When several users or teams share a cluster with a fixed number of nodes,
+there is a concern that one team could use more than its fair share of resources.
+
+Resource quotas are a tool for administrators to address this concern. Resource quotas
+work like this:
+- Different teams work in different namespaces.  Currently this is voluntary, but
+  support for making this mandatory via ACLs is planned.
+- Users put [compute resource limits](../user-guide/compute-resources.md) on their pods.
+- The administrator creates a Resource Quota for each namespace.
+- If users exceed the limits specified in the Resource Quota in a namespace, they are blocked
```
Comment (davidopp) [inline docs/admin/resource-quota.md]: This implies that they are allowed to exceed, and thereafter are blocked. But more generally, this is really vague.  This is in the admin section so we should explain in reasonable detail how this works. The pod gets created but the scheduler is not allowed to schedule it? If so, what prevents the scheduler from scheduling it?

File: docs/admin/resource-quota.md [hunk 2]
```diff
@@ -31,12 +31,30 @@ Documentation for other releases can be found at
...
+Examples of policies that could be created using namespaces and quotas are:
+- In a cluster with a capacity of 32 GiB RAM, and 16 cores, let team A use 20 Gib and 10 cores,
+  let B use 10GiB and 4 cores, and hold 2GiB and 2 cores in reserve for future allocation.
+- Limit the "testing" namespace to using 1 core and 1GiB RAM.  Let the "production" namespace
+  use any amount.
+
+In the case where the total capacity of the cell is less than the sum of the quotas of the namespaces,
+there may be contention for resources.  This is handled on a first-come-first-served basis.
+
+Neither contention nor changes to quota will affect already running pods.
```
Comment (davidopp) [inline docs/admin/resource-quota.md]: already-running

File: docs/admin/resource-quota.md [hunk 3]
```diff
@@ -48,7 +66,24 @@ Resource Quota is enforced in a particular namespace when there is a
...
-See [ResourceQuota design doc](../design/admission_control_resource_quota.md) for more information.
+## Compute Resource Quota
+
+The total sum of [compute resources](../user-guide/compute-resources.md) requested by pods
+in a namespace can be limited.  The following compute resource types are supported:
+
+| ResourceName | Description |
+| ------------ | ----------- |
+| cpu | Total cpu limits of containers |
+| memory | Total memory usage limits of containers
```
Comment (davidopp) [inline docs/admin/resource-quota.md]: memory usage -> memory

File: docs/admin/resource-quota.md [hunk 4]
```diff
@@ -128,22 +149,31 @@ services                3       5
 
 ## Quota and Cluster Capacity
 
-Resource Quota objects are independent of the Cluster Capacity.  They are
-expressed in absolute units.
+Resource Quota objects are independent of the Cluster Capacity. They are
+expressed in absolute units.  So, if you add nodes to your cluster, this does *not*
+automatically give each namespace the ability to consume resources.
```
Comment (davidopp) [inline docs/admin/resource-quota.md]: consumer resources -> consume more resources

File: docs/admin/resource-quota.md [hunk 5]
```diff
@@ -128,22 +149,31 @@ services                3       5
...
   - allow each tenant to grow resource usage as needed, but have a generous
     limit to prevent accidental resource exhaustion.
+  - detect demand from one namespace, add nodes, and increase quota.
 
 Such policies could be implemented using ResourceQuota as a building-block, by
 writing a 'controller' which watches the quota usage and adjusts the quota
-hard limits of each namespace.
+hard limits of each namespace according to other signals. 
+
+Note that resource quota divides up abstract resources, but it creates no restrictions around
```
Comment (davidopp) [inline docs/admin/resource-quota.md]: abstract resources -> aggregate resource consumption ?

File: docs/admin/resource-quota.md (hunk 1 above)
Comment (erictune) [inline docs/admin/resource-quota.md]: It 403 FORBIDDEN and not created.  Updated doc.

File: docs/admin/resource-quota.md (hunk 2 above)
Comment (erictune) [inline docs/admin/resource-quota.md]: done

File: docs/admin/resource-quota.md (hunk 3 above)
Comment (erictune) [inline docs/admin/resource-quota.md]: fixed.

File: docs/admin/resource-quota.md (hunk 4 above)
Comment (erictune) [inline docs/admin/resource-quota.md]: fixed.

File: docs/admin/resource-quota.md (hunk 5 above)
Comment (erictune) [inline docs/admin/resource-quota.md]: fixed.

**Comment by erictune** [comment]
@davidopp please re-review

**Comment by lavalamp** [comment]
LGTM rubber stamp.

**Comment by davidopp** [comment]
Reviewing now

File: docs/admin/resource-quota.md [hunk 6]
```diff
@@ -31,12 +31,35 @@ Documentation for other releases can be found at
...
+  status code `403 FORBIDDEN`.  Hint: Use the LimitRange admission controller to force default
+  values of limits before the quota is checked to avoid this problem.
+
+Examples of policies that could be created using namespaces and quotas are:
+- In a cluster with a capacity of 32 GiB RAM, and 16 cores, let team A use 20 Gib and 10 cores,
+  let B use 10GiB and 4 cores, and hold 2GiB and 2 cores in reserve for future allocation.
+- Limit the "testing" namespace to using 1 core and 1GiB RAM.  Let the "production" namespace
+  use any amount.
+
+In the case where the total capacity of the cell is less than the sum of the quotas of the namespaces,
```
Comment (davidopp) [inline docs/admin/resource-quota.md]: cell -> cluster

File: docs/admin/resource-quota.md [hunk 7]
```diff
@@ -48,7 +71,24 @@ Resource Quota is enforced in a particular namespace when there is a
...
+| cpu | Total cpu limits of containers |
+| memory | Total memory limits of containers
+| `example.com/customresource` | Total of `resources.limits."example.com/customresource"` of containers |
+
+For example, `cpu` quota sums up the `resources.limits.cpu` fields of every
+container of every pod in the namespace, and enforces a maximum on that sum.
+
+Any resource that is not part of core Kubernetes must follow the resource naming convention prescribed by Kubernetes.
+
+This means the resource must have a fully-qualified name (i.e. mycompany.org/shinynewresource)
```
Comment (davidopp) [inline docs/admin/resource-quota.md]: I'd remove the paragraph break between this sentence and the previous one.

File: docs/admin/resource-quota.md (hunk 1 above)
Comment (davidopp) [inline docs/admin/resource-quota.md]: I would clarify this as "If a new Pod would exceed the limits specified in the Resource Quota in a namespace, then creation of the Pod is blocked."

**Comment by davidopp** [comment]
Just a few last-minute comments.

File: docs/admin/resource-quota.md (hunk 7 above)
Comment (erictune) [inline docs/admin/resource-quota.md]: Deleted this line cause it turns out that arbitrary resources are not supported by current code.

**Pull request merged**